File: VB/WebApplication1/CustomControls/CustomAppointmentFormTemplateContainer .vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal

Namespace WebApplication1.CustomControls
	Public Class CustomAppointmentFormTemplateContainer
		Inherits AppointmentFormTemplateContainer
		Public Sub New(ByVal control As ASPxScheduler)
			MyBase.New(control)

		End Sub

		Public ReadOnly Property FormController() As AppointmentFormController
			Get
				Return Controller
			End Get
		End Property

		Public ReadOnly Property CustomInfoField() As String
			Get
				Return System.Convert.ToString(Appointment.CustomFields("AppointmentInfo"))
			End Get
		End Property

		Public ReadOnly Property CustomAttachment() As Byte()
			Get
				Return TryCast(Appointment.CustomFields("AppointmentAttachment"), Byte())
			End Get
		End Property

		Public ReadOnly Property CustomAttachmentFileName() As String
			Get
				If Appointment.CustomFields("AppointmentAttachmentFileName") Is Nothing Then
					Return ""
				Else
					Return "Uploaded file: " & System.Convert.ToString(Appointment.CustomFields("AppointmentAttachmentFileName"))
				End If
			End Get
		End Property


		Public ReadOnly Property Description() As String
			Get
				Return Appointment.Description
			End Get
		End Property
	End Class
End Namespace

File: CS/WebApplication1/CustomControls/CustomAppointmentFormTemplateContainer .cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;

namespace WebApplication1.CustomControls {
    public class CustomAppointmentFormTemplateContainer : AppointmentFormTemplateContainer {
        public CustomAppointmentFormTemplateContainer(ASPxScheduler control) : base(control) {            
        
        }

        public AppointmentFormController FormController { get { return Controller; } }

        public string CustomInfoField {
            get {
                return System.Convert.ToString(Appointment.CustomFields["AppointmentInfo"]);
            }
        }

        public byte[] CustomAttachment {
            get {
                return Appointment.CustomFields["AppointmentAttachment"] as byte[];
            }
        }

        public string CustomAttachmentFileName {
            get {
                if(Appointment.CustomFields["AppointmentAttachmentFileName"] == null)
                    return "";
                else
                    return "Uploaded file: " + System.Convert.ToString(Appointment.CustomFields["AppointmentAttachmentFileName"]);
            }
        }


        public string Description {
            get {
                return Appointment.Description;
            }
        }
    }
}